### DISCUSSION TITLE
Controlling LED on Feather 32u4 board

### DISCUSSION THREAD

**Discussion by ericbieri** [ghoti57/evofw3#36] [General]
Hi, first of all thanks for your great project. It helped me a lot!

I'm using a Adafruit Feather 32u4 ([Schematics](https://learn.adafruit.com/assets/28445)) together with a CC1101 Wing I recently designed myself. This works great with your code.

The only annoying thing is the LED of the feather, which is always on. I'd like to configure it such that it blinks when receiving messages. Unfortunately I don't have much experience with the AVR MCUs. 

According to the schematics the red LED is connected to pin PC7.

I therefore added the following lines 
#define HAS_LED
#define LED_DDR   DDRC
#define LED_PORT  PORTC
#define LED_PIN   7

here ..[https://github.com/ghoti57/evofw3/blob/85d02815f9513160ad6e5058fe994dbf5982c533/atm32u4_pins.h#L111-L113](https://github.com/ghoti57/evofw3/blob/85d02815f9513160ad6e5058fe994dbf5982c533/atm32u4_pins.h#L111-L113)

The LED is still always on. 

How/where do I have to configure the LED to achieve this?

Cheers Eric

**Comment by ghoti57** [comment]
It's great that you've got the code to work with your own hardware.

The reason that you can't get the code to do anything with the LED is there is nothing in the code to control it😁

None of the hardware I've used has had a led available to control,. They've all been connected to the host interfaces in some way.

It shouldn't be difficult to add some code to control the led to indicate it is receiving data.  The challenge will be to do this in a flexible way if you think it should be added to GitHub.

The design philosophy is not to have to edit the code itself but to use the custom board definitions as a pseudo makefile (to get round the Arduino gui's limitations)

You obviously have a good understanding of the hardware.  What are your coding skills like?

Pete

⁣Get BlueMail for Android ​

**Reply by ericbieri** [reply]
Hi Pete, thanks for your reply. My coding skills on AVR w. Arduino IDE are rather limited. I'll give it a shot as soon as I find time.  I mainly use ARM or ESP32 MCUs together with the PlatformIO IDE, which is great and comes with debugging for most MCUs. The CC1101 Featherwing is quite simple. I'll put it on my github as soon as V2 is ready. 
Have you got plans to port your code to an ARM-based device e.g. STM32 or ESP32? This would open up so many doors, e.g. forwarding the messages to an MQTT broker etc.

**Comment by ghoti57** [comment]
I'm thinking about an esp32 with MQTT as well as the existing serial interface.

⁣Get BlueMail for Android ​

**Reply by davidcreager** [reply]
Ooooh...that would be awesome!!!
